From 5ad1b92ddf7a2acb7c484590731e4e9eb99325f7 Mon Sep 17 00:00:00 2001
From: Kotresh HR <khiremat@redhat.com>
Date: Tue, 22 Dec 2015 12:29:32 +0530
Subject: [PATCH 210/214] geo-rep: Fix getting subvol count

Tiering doesn't support disperse volume as hot tier,
hence xml output doesn't give 'hotdisperseCount'.
Remove the usage of 'hotdisperseCount' in geo-rep
and return 0 instead.

BUG: 1291195
Change-Id: I283a0b60cfcfc82e731818aec47e3132b1bd2bb9
Signed-off-by: Kotresh HR <khiremat@redhat.com>
Reviewed-on: http://review.gluster.org/13062
Reviewed-on: http://review.gluster.org/13068
Tested-by: NetBSD Build System <jenkins@build.gluster.org>
Tested-by: Gluster Build System <jenkins@build.gluster.com>
Reviewed-by: Aravinda VK <avishwan@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/64379
Reviewed-by: Aravinda Vishwanathapura Krishna Murthy <avishwan@redhat.com>
Tested-by: Aravinda Vishwanathapura Krishna Murthy <avishwan@redhat.com>
---
 geo-replication/syncdaemon/monitor.py |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/geo-replication/syncdaemon/monitor.py b/geo-replication/syncdaemon/monitor.py
index 2d89859..5d66e8a 100644
--- a/geo-replication/syncdaemon/monitor.py
+++ b/geo-replication/syncdaemon/monitor.py
@@ -146,7 +146,10 @@ class Volinfo(object):
 
     def disperse_count(self, tier, hot):
         if (tier and hot):
-            return int(self.get('hotBricks/hotdisperseCount')[0].text)
+            # Tiering doesn't support disperse volume as hot brick,
+            # hence no xml output, so returning 0. In case, if it's
+            # supported later, we should change here.
+            return 0
         elif (tier and not hot):
             return int(self.get('coldBricks/colddisperseCount')[0].text)
         else:
-- 
1.7.1

